app/src/main/java/com/khmelenko/lab/varis/log/TextLeaf.java

```diff
@@ -0,0 +1,148 @@
+package com.khmelenko.lab.varis.log;
+
+import android.support.annotation.VisibleForTesting;
+
+public class TextLeaf extends LogEntryComponent {
+    @VisibleForTesting
+    Integer textColor;
+    @VisibleForTesting
+    Integer background;
+    @VisibleForTesting
+    boolean bold;
+    @VisibleForTesting
+    boolean italic;
+    @VisibleForTesting
+    boolean underline;
+    public String text = "";
+
+    TextLeaf() {
+    }
+
+    public TextLeaf(String text) {
+        this.text = text;
+    }
+
+    TextLeaf(TextLeaf textLeaf) {
+        textColor = textLeaf.textColor;
+        background = textLeaf.background;
+        bold = textLeaf.bold;
+        italic = textLeaf.italic;
+        underline = textLeaf.underline;
+    }
+
+    void addAnsiCodes(String[] ansiStates) {
+        for (String ansiCode : ansiStates) {
+            translateAnsiCode(ansiCode);
+        }
+    }
+
+    private void translateAnsiCode(String code) {
```

Comment:
This function looks more like utililities function, I would move it to once of the utilities classes

Reply:
You are right, I would move it to the AnsiParser class if we don't move the colors to colors.xml